OpenSim/Simulation/Model/PathPoint.h

```diff
@@ -86,68 +65,106 @@ OpenSim_DECLARE_CONCRETE_OBJECT(PathPoint, Object);
     // CONSTRUCTION
     //--------------------------------------------------------------------------
 public:
-    PathPoint();
-    PathPoint(const PathPoint &aPoint);
-    virtual ~PathPoint();
+    PathPoint() : Station() {}
 
-#ifndef SWIG
-    PathPoint& operator=(const PathPoint &aPoint);
-#endif
+    virtual void init(const PathPoint& point) {
+        *this = point;
+        copyData(point);
+    }
    void copyData(const PathPoint &aPoint);
-    virtual void init(const PathPoint& aPoint);
 
 #ifndef SWIG
-    const SimTK::Vec3& getLocation() const { return _location; }
+    const SimTK::Vec3& getLocation() const { return get_location(); }
 #endif
-    SimTK::Vec3& getLocation()  { return _location; }
+    SimTK::Vec3& getLocation()  { return upd_location(); }
+
+    const double& getLocationCoord(int aXYZ) const {
+        assert(aXYZ>=0 && aXYZ<=2); return get_location()[aXYZ];
+    }
 
-    const double& getLocationCoord(int aXYZ) const { assert(aXYZ>=0 && aXYZ<=2); return _location[aXYZ]; }
-    void setLocationCoord(int aXYZ, double aValue) { assert(aXYZ>=0 && aXYZ<=2); _location[aXYZ]=aValue; }
     // A variant that uses basic types for use by GUI
+    void setLocationCoord(int aXYZ, double aValue) {
+        assert(aXYZ>=0 && aXYZ<=2);
+        upd_location()[aXYZ]=aValue;
+    }
+
+    void setLocation( const SimTK::State& s, const SimTK::Vec3& location) {
+        // TODO this is completely wrong to be taking a state and changing a property!
+        upd_location() = location;
+    }
+    void setLocation( const SimTK::State& s, int aCoordIndex, double aLocation) {
+        if (aCoordIndex >= 0 && aCoordIndex <= 2)
+           upd_location()[aCoordIndex] = aLocation;
+    }
 
-    void setLocation( const SimTK::State& s, const SimTK::Vec3& aLocation);
-    void setLocation( const SimTK::State& s, int aCoordIndex, double aLocation);
-    void setLocation( const SimTK::State& s, double pt[]){ // A variant that uses basic types for use by GUI
+    // A variant that uses basic types for use by GUI
+    void setLocation( const SimTK::State& s, double pt[]){ 
         setLocation(s,SimTK::Vec3::updAs(pt));
     }
-    void setBody(const PhysicalFrame& aBody);
-    void changeBodyPreserveLocation(const SimTK::State& s, PhysicalFrame& aBody);
+    void setBody(const PhysicalFrame& body) {
+        this->setParentFrame(body);
+    }
+
+    void changeBodyPreserveLocation(const SimTK::State& s, PhysicalFrame& body){
```

Comment:
Yes, this method is useful for GUI operation since it keeps the point at same 3D location while changing Body/PhysicalFrame.